sdk-extensions/autoconfigure/src/main/java/io/opentelemetry/sdk/autoconfigure/DefaultConfigProperties.java

```diff
@@ -56,7 +56,7 @@ private DefaultConfigProperties(
         (name, value) -> config.put(name.toLowerCase(Locale.ROOT).replace('_', '.'), value));
     systemProperties.forEach(
         (key, value) ->
-            config.put(((String) key).toLowerCase(Locale.ROOT).replace('-', '.'), (String) value));
+            config.put(key.toString().toLowerCase(Locale.ROOT).replace('-', '.'), (String) value));
```

Comment:
```suggestion
            config.put(key.toString().toLowerCase(Locale.ROOT).replace('-', '.'), value.toString()));
```

Reply:
right better to convert the value as well. fixed thanks